src/EditorFeatures/CSharp/EventHookup/EventHookupSessionManager_EventHookupSession.cs

```diff
@@ -106,27 +107,21 @@ public EventHookupSession(
 
             var asyncToken = asyncListener.BeginAsyncOperation(GetType().Name + ".Start");
 
-            _eventNameTask = Task.Factory.SafeStartNewFromAsync(
-                () => DetermineIfEventHookupAndGetHandlerNameAsync(document, position, cancellationToken),
-                cancellationToken,
-                TaskScheduler.Default);
```

Comment:
instead of kicking off a task here, we just make DetermineIfEventHookupAndGetHandlerNameAsync switch to TaskSchedule.Default as the first thing it does.